src/Data/Functor/Invariant.hs

```diff
@@ -986,3 +998,19 @@ these, you can derive them using the "Data.Functor.Invariant.TH" module.
 genericInvmap :: (Generic1 f, Invariant (Rep1 f)) => (a -> b) -> (b -> a) -> f a -> f b
 genericInvmap f g = to1 . invmap f g . from1
 #endif
+
+-------------------------------------------------------------------------------
+-- Wrappers
+-------------------------------------------------------------------------------
+
+-- | 'Profunctor' with the same input and output args can be seen as an 'Invariant' functor.
+newtype InvariantProfunctor p a = InvariantProfunctor (p a a)
+
+instance Profunctor p => Invariant (InvariantProfunctor p) where
+  invmap fn1 fn2 (InvariantProfunctor f) = InvariantProfunctor (dimap fn2 fn1 f)
```

Comment:
Since you've defined `invmapProfunctor` now, you might as well use it to avoid code duplication:

```suggestion
  invmap = invmapProfunctor
```

Reply:
Oops, I overlooked the fact that the types don't quite line up here the way they do for `WrappedFunctor` and friends. In that case, please ignore this suggestion :)